### DISCUSSION TITLE
How to see the final resolved renovate config with several presets?

### DISCUSSION THREAD

**Discussion by dirkw-of** [renovatebot/renovate#26404] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.105.3

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps Server

### What is your question?

I am creating several presets which I include in a default.json preset, but for one preset file my rules aren't picked up properly, whereas the rules are picked up when they are in the default.json file directly.

Before posting all the details here to ask for more help, I am wondering if there is a way to see what the final configuration json looks like after all presets have been resolved/inserted? Then it's a lot easier to determine why my configuration isn't behaving as I expect it to.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Try setting printConfig=true

**Reply by dirkw-of** [reply]
And as environment variable:
RENOVATE_PRINT_CONFIG = true
